tests/conftest.py

```diff
@@ -0,0 +1,156 @@
+        'docker',
+        'run',
+        '-d',
+        '-p',
+        f'127.0.0.1:{port}:5000',
+        '--rm',
+        '--name',
+        fake_registry_name,
+        'registry'
+    )
+    subprocess.check_call(run_registry_command)
+    registry_uri = f'127.0.0.1:{port}'
+    _check_registry_is_up(registry_uri)
+    yield registry_uri
+    subprocess.check_call(('docker', 'rm', '-f', fake_registry_name))
+
+
+@pytest.fixture(scope='session')
+def fake_image_foo_name():
+    image_name = _build_testing_image('foo')
+    yield image_name
+    _delete_image(image_name)
+
+
+@pytest.fixture(scope='session')
+def fake_image_bar_name():
+    image_name = _build_testing_image('bar')
+    yield image_name
+    _delete_image(image_name)
+
+
+@pytest.fixture
+def dummy_deb_nginx():
+    image_name = _build_testing_image('dummy_deb_nginx')
+    port = reserve()
+    dummy_deb_nginx_name = _get_name_with_random_suffix('dummy_deb_nginx')
+    run_dummy_deb_nginx_command = (
+        'docker',
+        'run',
+        '-d',
+        '-p',
+        f'127.0.0.1:{port}:80',
+        '--rm',
+        '--name',
+        dummy_deb_nginx_name,
+        image_name
+    )
+    subprocess.check_call(run_dummy_deb_nginx_command)
+    image_ip = _inspect_image(dummy_deb_nginx_name)[
+        'NetworkSettings']['Networks']['bridge']['IPAddress']
+    yield (dummy_deb_nginx_name, image_ip)
+    _delete_image(image_name)
+
+
+@pytest.fixture
+def fake_baz_dummy_deb_images(tmpdir, dummy_deb_nginx):
+    nginx_name, nginx_ip = dummy_deb_nginx
+    baz_path = _create_baz_dockerfile(nginx_ip, tmpdir)
+    baz_dummy_deb_image_name = _build_testing_image(
+        'baz',
+        dockerfile_path=baz_path,
+        with_no_cache=True
+    )
+    subprocess.check_call(('docker', 'stop', nginx_name))
+    baz_no_dummy_deb_image_name = _build_testing_image(
+        'baz',
+        dockerfile_path=baz_path,
+        with_no_cache=True
+    )
+    yield baz_dummy_deb_image_name, baz_no_dummy_deb_image_name
+    _delete_image(baz_dummy_deb_image_name)
+    _delete_image(baz_no_dummy_deb_image_name)
+
+
+def _build_testing_image(
+    image_name,
+    dockerfile_path=None,
+    with_no_cache=False,
+):
+    randomized_image_name = _get_name_with_random_suffix(image_name)
+    if not dockerfile_path:
+        dockerfile_path = f'{DEFAULT_TESTING_IMAGES_PATH}/{image_name}'
+    build_command = (
+        'docker', 'build', dockerfile_path, '-t', randomized_image_name
+    )
+    if with_no_cache:
+        build_command += ('--no-cache',)
+    subprocess.check_call(build_command)
+    return randomized_image_name
+
+
+def _get_name_with_random_suffix(name):
+    return '{}_{}'.format(
+        name,
+        ''.join(random.choices(string.ascii_lowercase + string.digits, k=6)),
+    )
+
+
+def _delete_image(image_tag):
+    image_id = _inspect_image(image_tag)['Id']
+    subprocess.check_call(('docker', 'rmi', '-f', image_id))
+
+
+def _check_registry_is_up(registry_uri):
```

Comment:
a better name: `_wait_for_registry` -- also, idk why you made this a function, just inline it (your code is much more difficult to read when I have to keep jumping around for functions which are only used once!)

Reply:
This isolates all of the healthcheck logic to a unit and documents it with the function name. I tend to leave all of the "public" pytest fixtures near the top since that's usually what one cares about when looking at conftest files, but that does make this helper function stray really far from where it's used...

I'm leaning towards leaving it as is, but if you feel against it still I could move/remove it